### ISSUE TITLE
progress bar moves upwards in screen 

### ISSUE THREAD

**Issue by saj1919** [tqdm/tqdm#798]
When I run for loop with tqdm in terminal ... it works fine.
When I run the same thing in linux `screen` ... below thing happens.
I think its a **bug**. Any code with tqdm around for loop can be checked for the replication.
Was able to replicate in Ubuntu and Mac

Steps :
1. screen -Rd test_screen
2. run python script with tqdm for loop
Thats it.

<img width="1422" alt="Screenshot 2019-08-22 at 5 44 17 PM" src="https://user-images.githubusercontent.com/8842245/63513719-957c4080-c504-11e9-9b9e-a19b581b74e1.png">

**Comment by casperdcl** [comment]
duplicate of #336?

Looks like you could fix it with `ncols=150` or similar, screen seems to be reporting too many `ncols`.

**Comment by saj1919** [comment]
It is not a direct duplicate of #336 as here progress-bar is moving upwards which is worse.
In case you press `Enter` it comes down as usual and then again keeps going upwards. Hard to keep track of whats running because normal print statements will print as usual over these progress-bar once tqdm is finished.

I tried `ncols=150` ... didn't work.

**Comment by casperdcl** [comment]
so with `ncols=150`, the bar doesn't fill the full width of the screen but you still see it moving upwards?

**Comment by saj1919** [comment]
No different output even after choosing `ncols=150` ... it moved upwards still.

**Comment by casperdcl** [comment]
right. judging by the fact that it works fine in your terminal outside `screen`, though, I'm hoping it's a bug with `screen`.

There's no way a single bar should be moving upwards. Nested ones; perhaps.

Can you try:

```python
print('a\nb\rc\nd\re')
```

the output should be

```
a
c
e
```

**Comment by saj1919** [comment]
Hi, sorry for delayed answer.
Output ->
```
a
c
e
```
Were you able to reproduce this issue on Ubuntu screens ?
For me it happens every time. And my use case is to run my code in screen and see the progress. Now I am using nohup instead of screen just to see the progress.

**Comment by casperdcl** [comment]
- I use `tmux` which seems to work without issue
- you may need `screen -U` (https://stackoverflow.com/questions/52250556/tqdm-in-screen-environment-printing-new-line-and-unknown-charactors)

**Comment by stoyian** [comment]
based on this [post](https://stackoverflow.com/questions/52250556/tqdm-in-screen-environment-printing-new-line-and-unknown-charactors) you need to enable UTF-8 encoding. To do that simply run `screen -U`
It worked for me at least 😃

**Comment by xiezhipeng-git** [comment]
I have same problem with windows wsl,And I found,If change shell width large. it is not move.if width small.it is move.change ncols do not work.is it any other function to slove it.

**Comment by FuryMartin** [comment]
> duplicate of #336?
> 
> Looks like you could fix it with `ncols=150` or similar, `screen` seems to be reporting too many `ncols`.

This works for me. I set `ncols=100` and it works fine.
